ts-client/src/ValorClient.ts

```diff
@@ -0,0 +1,337 @@
+import axios, { AxiosInstance } from 'axios';
+
+type Dataset = {
+  id: number;
+  name: string;
+  metadata: object;
+};
+
+type Model = {
+  id: number;
+  name: string;
+  metadata: object;
+};
+
+type TaskType =
+  | 'skip'
+  | 'empty'
+  | 'classification'
+  | 'object-detection'
+  | 'semantic-segmentation'
+  | 'embedding';
+
+type Evaluation = {
+  id: number;
+  model_name: string;
+  datum_filter: object;
+  parameters: { task_type: TaskType; object };
+  status: 'pending' | 'running' | 'done' | 'failed' | 'deleting';
+  metrics: object[];
+  confusion_matrices: object[];
+};
+
+const metadataDictToString = (input: { [key: string]: string | number }): string => {
+  const result: { [key: string]: Array<{ value: string | number; operator: string }> } =
+    {};
+
+  Object.entries(input).forEach(([key, value]) => {
+    result[key] = [{ value: value, operator: '==' }];
+  });
+
+  return JSON.stringify(result);
+};
+
+export class ValorClient {
+  private client: AxiosInstance;
+
+  /**
+   *
+   * @param baseURL - The base URL of the Valor server to connect to.
+   */
+  constructor(baseURL: string) {
+    this.client = axios.create({
+      baseURL,
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    });
+  }
+
+  /**
+   * Fetches datasets matching the filters defined by queryParams. This is private
+   * because we define higher-level methods that use this.
+   *
+   * @param queryParams An object containing query parameters to filter datasets by.
+   *
+   * @returns {Promise<Dataset[]>}
+   *
+   */
+  private async getDatasets(queryParams: object): Promise<Dataset[]> {
+    const response = await this.client.get('/datasets', { params: queryParams });
+    return response.data;
```

Comment:
```suggestion
    return this.client.get('/datasets', { params: queryParams }).data;
```
await is implied when returning from an async function. if the call fails, the promise will reject basically the same as it would've.